app/src/main/java/com/example/taskthree/view/GuessFragment.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.taskthree.view
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import com.example.taskthree.R
+import com.example.taskthree.databinding.FragmentCharBinding
+import com.example.taskthree.viewmodel.GuessViewModel
+
+class GuessFragment : Fragment() {
+    private lateinit var binding: FragmentCharBinding
+    private val viewModel: GuessViewModel by activityViewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = FragmentCharBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        buttonConnectEditText()
+        viewModel.resultMessageObs.observe(viewLifecycleOwner) { message ->
+            if (message == getString(R.string.try_again_message)) {
+                binding.textViewGuessHint.text = viewModel.randomCharObs.toString()
+            } else if (message == getString(R.string.win_message)) {
+                binding.textViewGuessHint.text =
+                    StringBuilder().append(getString(R.string.guess_character_hint, viewModel.randomIntObs.value))
```

Comment:
Buradaki duzenleme anlasilmamis sanirim. StringBuilder kullanimina gerek olmadan strings.xml deki string'ine parametre ekleyebiliyorsun;
<string name="welcome_messages">Hello, %1$s! You have %2$d new messages.</string>
var text = getString(R.string.welcome_messages, username, mailCount)

Append ile ugrasmaya gerek kalmayacak bu durumda

Reply:
Şimdi anladım hocam teşekkür ederim